x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality_panel/data_quality_details/indices_details/pattern/index.tsx

```diff
@@ -242,92 +255,87 @@ const PatternComponent: React.FC<Props> = ({
     httpFetch,
     formatBytes,
     formatNumber,
+    flyoutChartSelectedActionAbortControllerRef,
   ]);
 
-  useEffect(() => {
-    const flyoutIndexExpandActionAbortController =
-      flyoutIndexExpandActionAbortControllerRef.current;
-    const tableRowIndexCheckNowActionAbortController =
-      tableRowIndexCheckNowActionAbortControllerRef.current;
-    const flyoutIndexChartSelectedActionAbortController =
-      flyoutIndexChartSelectedActionAbortControllerRef.current;
-    return () => {
-      flyoutIndexExpandActionAbortController.abort();
-      tableRowIndexCheckNowActionAbortController.abort();
-      flyoutIndexChartSelectedActionAbortController.abort();
-    };
-  }, []);
-
   return (
     <div data-test-subj={`${pattern}PatternPanel`}>
-      <PatternAccordion
-        id={patternComponentAccordionId}
-        initialIsOpen={true}
-        buttonElement="div"
-        buttonContent={
-          <PatternSummary
-            incompatible={getTotalPatternIncompatible(patternRollup?.results)}
-            indices={indexNames?.length}
-            indicesChecked={getTotalPatternIndicesChecked(patternRollup)}
-            ilmExplainPhaseCounts={ilmExplainPhaseCounts}
-            pattern={pattern}
-            patternDocsCount={patternRollup?.docsCount ?? 0}
-            patternSizeInBytes={patternRollup?.sizeInBytes}
-          />
-        }
+      <HistoricalResultsContext.Provider
+        value={{
+          fetchHistoricalResults,
+          historicalResultsState,
+        }}
```

Comment:
Would it make sense to memoize this object? this provider is receiving a new value every update

Reply:
1. I agree we can add it for optimization.

2.  It's only the identity of a wrapper that is changing, values themselves are not. And application is using the values. It does trigger rerender, but rerender is generally not a problem unless you misuse useeffect and trigger unsafe sideeffects from there which is a bad architecture from the get go.

Reply:
Wrapped in useMemo with latest commit.

Reply:
Great thanks for addressing it